src/github.ts

```diff
@@ -1916,13 +1916,37 @@ export class GitHub {
     return protectionRule as GraphQLLockBranchProtectionRule;
   }
 
+  async isBranchSyncedWithPullRequestCommits(
+    branchName: string,
+    pullRequest: PullRequest
+  ): Promise<boolean> {
+    const prCommits = await this.request(
+      'GET /repos/{owner}/{repo}/pulls/{pull_number}/commits',
+      {
+        pull_number: pullRequest.number,
+        owner: this.repository.owner,
+        repo: this.repository.repo,
+      }
+    );
```

Comment:
This is fantastic, I found out today about this endpoint by pure serendipity. And that's exactly what we want :)